src/commcare_cloud/fab/deploy_diff.py

```diff
@@ -59,7 +58,7 @@ def get_diff_context(self):
             context["errors"].append("Versions are identical. No changes since last deploy.")
             return context
 
-        if not (github_auth_provided() and self.current_commit and self.deploy_commit):
+        if not (self.current_commit and self.deploy_commit):
```

Comment:
If auth isn't provided, github restricts you to [60 requests per hour](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting).  Since this fetches each PR individually, there's a high likelihood of hitting this limit.  I'm not sure what that failure mode looks like, but I think the code should either require you to provide auth or (preferably) skip the diff and warn you to add auth if you want to see it.

Reply:
504a2aaae